healthvault-ios-sdk/Classes/Controllers/MHVBrowserAuthBroker.m

```diff
@@ -91,85 +94,84 @@ - (void)showAuthenticationControllerFromViewController:(UIViewController *)viewC
             vc = [[[UIApplication sharedApplication] keyWindow] rootViewController];
         }
         
-        UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:self];
+        if (!self.browserController)
+        {
+            self.browserController = [[MHVBrowserController alloc] initWithAuthBroker:self];
+        }
+        
+        UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:self.browserController];
         
         [vc presentViewController:navigationController animated:YES completion:nil];
     }];
 }
 
 - (void)completeWithSuccessUrl:(NSURL *)successUrl error:(NSError *)error
 {
-    [self dismissViewControllerAnimated:YES completion:^
+    [self.browserController dismissViewControllerAnimated:YES completion:^
      {
          dispatch_async(self.authQueue, ^
                         {
-                            self.target = nil;
+                            self.startUrl = nil;
                             self.endUrl = nil;
-                            self.completion(successUrl, error);
-                            self.completion = nil;
+                            self.browserController = nil;
+                            
+                            if (self.completion)
+                            {
+                                self.completion(successUrl, error);
+                                self.completion = nil;
+                            }
                         });
      }];
 }
 
-#pragma mark - Overrides
-
-- (void)viewWillDisappear: (BOOL)animated
-{
-    // Overriding default behavior of MHVBrowserController
-    [super viewWillDisappear:animated];
-}
-
-- (void)abort
-{
-    // Overriding default behavior of MHVBrowserController
-}
-
-- (void)cancelButtonPressed:(id)sender
+- (void)userCancelled
 {
-    [self completeWithSuccessUrl:nil error:[NSError error:[NSError MHVOperationCancelled] withDescription:@"The operation has been cancelled by the user."]];
+    [self completeWithSuccessUrl:nil
+                           error:[NSError error:[NSError MHVOperationCancelled] withDescription:@"The operation has been cancelled by the user."]];
 }
 
-#pragma mark - UIWebViewDelegate
+#pragma mark - WKWebViewDelegate
```

Comment:
I think we need to add in didReceiveAuthenticationChallenge:, didFailProvisionalNavigation:, and webViewWebContentProcessDidTerminate:. I want to make sure that we don't orphan this completion block.